### PR TITLE
fix(text-input): fix disabled text on dark themes in safari

### LINKED ISSUE #6673: Safari's input[disabled] text is too faded
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

Safari's input[disabled] text is too faded

> Is this issue related to a specific component?

TextInput

> What browser are you working in?

Safari Version 13.1.1
Chrome Version 83

> What version of the Carbon Design System are you using?

7.17.0

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

## Steps to reproduce the issue

`
<TextInput disabled value="value" labelText="Label text" />
`

## Additional information

Similar issue https://github.com/jgthms/bulma/issues/2355

It looks like one solution is patching:

`
.bx--text-input {
-webkit-text-fill-color: currentcolor;
}
`

Screen Shot 2020-08-14 at 15 54 32

### PR DESCRIPTION
**Pull request by tw15egan**
Closes #6673 

Fixes an issued with disabled text in safari when using dark themes

#### Changelog

**New**

- WebKit specific fix for disabled text

#### Testing / Reviewing

Go to text input, change to `g90` or `g100` in the preview, and disabled the input

### CODE REVIEW

File: packages/components/src/components/text-input/_text-input.scss [hunk 1]
```diff
@@ -156,7 +156,8 @@
     color: $disabled-02;
     background-color: $disabled-01;
     border-bottom: 1px solid transparent;
-
+    // Needed to fix disabled text in Safari #6673
+    -webkit-text-fill-color: currentcolor;
```
Comment (joshblack) [inline packages/components/src/components/text-input/_text-input.scss]: Random, is casing significant here or should it be `currentColor`?

Comment (tw15egan) [reply packages/components/src/components/text-input/_text-input.scss]: Oh weird, I thought casing makes a difference but I guess it doesn't! Works as is but I'm going to fix it anyways for consistency 😉

**Review by emyarod — APPROVED** [review]
looks good to me

**Review by joshblack — APPROVED** [review]

File: packages/components/src/components/text-input/_text-input.scss [hunk 2]
```diff
@@ -156,7 +156,8 @@
     color: $disabled-02;
     background-color: $disabled-01;
     border-bottom: 1px solid transparent;
-
+    // Needed to fix disabled text in Safari #6673
+    -webkit-text-fill-color: currentColor;
```
Comment (oliviertassinari) [inline packages/components/src/components/text-input/_text-input.scss:160]: This is the initial value, it's a noop: https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-text-fill-color.

**Pull request merged**